chore(auth/shared): add `transactionId` to Apple subscriptions in Firestore

sardesam opened the pull request:
## Because

- Currently for Apple subscriptions the `originalTransactionId` is included, but not the `transactionId`.
- Including `transactionId` would provide an important indicator whether it's the initial transaction or a restore/renewal.

## This pull request

- Adds `transactionId` to Apple subscriptions in Firestore.

## Issue that this pull request solves

Closes: FXA-6987

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

biancadanforth reviewed (approved):
R+WC. Thanks Meghan; this LGTM and WFM locally. Just one small change requested to make `transactionId` a required field and reorder it among the properties. Can you update that before merging?

sardesam commented:
Thanks for clarifying, Bianca!

For visibility I'll share the Slack message I sent you asking this exact question in case this ever changes or we need to make updates and we have a reference that it was considered before.

I followed a similar pattern to what you had done in #14592 (also referenced in the Jira) because it was adding an additional field. I also made it `optional` (initially; updated to required now) because, while it is a unique identifier, its purpose is only to distinguish between initial subscription and subsequent renewals (as per the docs). Meaning, this string changes each time and if there are no subsequent renewals, it will always match `originalTransactionId` which is the unique identifier for the product/service.

My question was whether it should be changed to `required` since it will exist on all purchases (confirmed yes, and updated, since it is always present on `JWSTransactionDecodedPayload`). I didn't include it on other checks but wondered whether I should in case that logic can be leveraged in some way later?

I couldn't come up with a scenario because we don't manage IAP subscriptions and block upgrades (must be done from the PlayStore or App Store), so I wasn't sure if there really was any more information we needed or if this additional information can be leveraged at all on our end.
